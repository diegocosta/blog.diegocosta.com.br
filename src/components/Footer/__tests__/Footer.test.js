import React from 'react';
import { render } from '@testing-library/react';

import { Footer } from '../';

describe('<Footer />', () => {
  it('should render properly', () => {
    const contacts = [
      { link: 'mailto:johndoe@doe.com', label: 'email' },
      { link: 'https://google.com', label: 'site' },
    ];

    const { baseElement, queryByText } = render(<Footer year={2020} contacts={contacts} />);
    expect(queryByText('source code')).toBeFalsy();
    expect(baseElement).toMatchSnapshot();
  });

  it('should render the repository url if passed', () => {
    const { baseElement, getByText } = render(
      <Footer year={2020} contacts={[]} repository="https://github.com/diegocosta/blog.diegocosta.com.br" />
    );
    expect(getByText('source code')).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
