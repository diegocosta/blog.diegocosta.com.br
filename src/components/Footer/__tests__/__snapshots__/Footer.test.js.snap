// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render properly 1`] = `
.c0 {
  margin-bottom: 50px;
}

.c0 nav {
  margin-bottom: 50px;
}

.c1 {
  font-size: 16px;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-shadow: none;
}

.c2:hover,
.c2:focus {
  border-bottom: 1px solid;
  outline: none;
}

@media (min-width:760px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c0 nav {
    margin-bottom: 0;
  }
}

@media (min-width:760px) {
  .c1 {
    font-size: 18px;
  }
}

<body>
  <div>
    <footer
      class="c0"
    >
      <p>
        Navigation
      </p>
      <div
        class="c1"
      >
        <span
          class=""
        >
          © 
          2020
          , built with 
        </span>
        <a
          class="c2"
          href="https://gatsbyjs.org"
          rel="noopener"
          target="_blank"
        >
          gatsby
        </a>
        <span
          class=""
        >
           • 
        </span>
      </div>
    </footer>
  </div>
</body>
`;

exports[`<Footer /> should render the repository url if passed 1`] = `
<body>
  .c0 {
  margin-bottom: 50px;
}

.c0 nav {
  margin-bottom: 50px;
}

.c1 {
  font-size: 16px;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-shadow: none;
}

.c2:hover,
.c2:focus {
  border-bottom: 1px solid;
  outline: none;
}

@media (min-width:760px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c0 nav {
    margin-bottom: 0;
  }
}

@media (min-width:760px) {
  .c1 {
    font-size: 18px;
  }
}

<div>
    <footer
      class="c0"
    >
      <p>
        Navigation
      </p>
      <div
        class="c1"
      >
        <span
          class=""
        >
          © 
          2020
          , built with 
        </span>
        <a
          class="c2"
          href="https://gatsbyjs.org"
          rel="noopener"
          target="_blank"
        >
          gatsby
        </a>
        <span
          class=""
        >
           • 
        </span>
        <a
          class="c2"
          href="https://github.com/diegocosta/diegocosta.com.br"
          rel="noopener"
          target="_blank"
        >
          source code
        </a>
      </div>
    </footer>
  </div>
</body>
`;
